### ISSUE TITLE
Help Wanted: appendBuffer returning the end of the file, not the offset. 

### ISSUE THREAD

**Issue by ericwooley** [gpac/mp4box.js#53]
Im am converting something which works, https://github.com/ericwooley/StreamTor/blob/master/app/utils/videostream.js (line 142 works)

to a class which is more manipulatable.
https://github.com/ericwooley/StreamTor/blob/master/app/VideoQueue.js (line 165, is the problem)

The problem I am having is that mp4Box.append buffer returns the end of the file, instead of the offset. I can't figure out what is happening differently in the two files. Any help would be appreciated

**Comment by cconcolato** [comment]
Glad that you're using MP4Box. I'm assuming you are using the same buffers in both cases, right? The code looks the same to me. Could you provide logs? I have noticed some bugs in the nextOffset computation in some weird cases. Maybe you're hitting this problem. 

Also your code says: "MP4Box tends to blow up ungracefully when it can't parse the mp4 input". What do you mean by "when it can't parse the mp4 input" ? Is it a valid mp4 ? or are you passing garbage data? 

There's also: "// Prevent infinte loops if mp4box keeps requesting the same data". This should only happen when the end of the file has been reached or when you're passing data that does not reach the position that MP4box asked for. Is it the case? Is the issue solved since you've commented it out?

Nice ES6 code btw.

**Comment by ericwooley** [comment]
@cconcolato Thanks for getting back so quickly! I just added some comments to help with debugging, so the files are a little more updated

Here is the console output from my class
~~~
Array Buffer Append                                       
  Appending array buffer ArrayBuffer {} byteLength 16384  
  Next Offset 5543151                                     
mediaSource ready                                         
Array Buffer Append                                       
  Appending array buffer ArrayBuffer {} byteLength 11025  
  Next Offset 5554176
~~~
and here is the output from the videoStream library which is considerably longer, I took out a large chunk in the middle, where it seemed like the pattern was simply repeating, but if you want the full output i put it here http://pastebin.com/dyC9ME3P
~~~
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16384
  Next Offset 5543151
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 11025
  Next Offset 5554176
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 3968
  videoStream.js:48 MP4 info: Object {duration: 16672, timescale: 1000, isFragmented: false, fragment_duration: 0, isProgressive: false…}
  Next Offset 48
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16336
  Next Offset 16384
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16384
  Next Offset 16384
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16384
  Next Offset 32768
/********************************************************************
* A lot more of that pattern was here, but I trimmed it down,
* full output here http://pastebin.com/dyC9ME3P
*********************************************************************/
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16384
  Next Offset 5488640
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16384
  Next Offset 5505024
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16384
  Next Offset 5521408
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16384
  Next Offset 5537792
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 16384
  Next Offset 5558144
Array Buffer Append
  Appending array buffer ArrayBuffer {} byteLength 3968
  Next Offset 5558144
~~~

Also, I am using the same mp4 file in each example. https://www.dropbox.com/s/6xy11y90a1kwuqu/test-09-Jun-15-05%3A47%3A22-large.processed.mp4?dl=0 

If you want to debug the example yourself, just clone this https://github.com/ericwooley/StreamTor and run it, then http://localhost:8080/broadcast and drop that file on the area specified, there is a checkbox that switches which one gets used. However, you need to refresh the page between each use.

As for the comments, I shamefully stole this code from here https://github.com/jhiesey/videostream because it did *almost* exactly what I wanted it too, except depended on a specific fork of mp4box, and was very hard to follow. So most of those comments are copied over.

**Comment by cconcolato** [comment]
Are you still having this issue? I did not have time to check. Did you make any progress?

**Comment by ericwooley** [comment]
I ended up switching to webm, sorry, don't know if it's still an issue.
